test/config/config.ts

```diff
@@ -1,98 +1,346 @@
+import { execSync } from 'child_process';
 import type { Dialect, Options } from '@sequelize/core';
+import { default as semver } from 'semver';
 
 const { env } = process;
 
-export const Config: Record<Dialect, Options> = {
+//  ██████╗ ██╗ █████╗ ██╗     ███████╗ ██████╗████████╗███████╗
+//  ██╔══██╗██║██╔══██╗██║     ██╔════╝██╔════╝╚══██╔══╝██╔════╝
+//  ██║  ██║██║███████║██║     █████╗  ██║        ██║   ███████╗
+//  ██║  ██║██║██╔══██║██║     ██╔══╝  ██║        ██║   ╚════██║
+//  ██████╔╝██║██║  ██║███████╗███████╗╚██████╗   ██║   ███████║
+//  ╚═════╝ ╚═╝╚═╝  ╚═╝╚══════╝╚══════╝ ╚═════╝   ╚═╝   ╚══════╝
+//
+
+const configs: { [key in Dialect]: Record<('default' | 'native' | number), Options> } = {
+
+  //  ╔═╗╔═╗ ╦    ╔═╗╔═╗╦═╗╦  ╦╔═╗╦═╗  ┌─┐┌─┐┌┬┐┌┬┐┬┌┐┌┌─┐┌─┐
+  //  ╚═╗║═╬╗║    ╚═╗║╣ ╠╦╝╚╗╔╝║╣ ╠╦╝  └─┐├┤  │  │ │││││ ┬└─┐
+  //  ╚═╝╚═╝╚╩═╝  ╚═╝╚═╝╩╚═ ╚╝ ╚═╝╩╚═  └─┘└─┘ ┴  ┴ ┴┘└┘└─┘└─┘
+
   mssql: {
-    host: env.SEQ_MSSQL_HOST || env.SEQ_HOST || 'localhost',
-    username: env.SEQ_MSSQL_USER || env.SEQ_USER || 'SA',
-    password: env.SEQ_MSSQL_PW || env.SEQ_PW || 'Password12!',
-    port: env.SEQ_MSSQL_PORT || env.SEQ_PORT || 22_019,
-    database: env.SEQ_MSSQL_DB || env.SEQ_DB || 'sequelize_test',
-    dialectOptions: {
-      options: {
-        encrypt: false,
-        requestTimeout: 25_000,
+    // When SQL Server has one instance it runs under 1433, with multiple instances
```

Comment:
This looks much more complicated than it needs to be
If you're going to add support for having both oldest & latest run at the same time, I recommend you have an object that looks like this instead:

```ts
const configs: {
  base: {
    host: env.SEQ_MSSQL_HOST || env.SEQ_HOST || 'localhost',
    username: env.SEQ_MSSQL_USER || env.SEQ_USER || 'SA',
    password: env.SEQ_MSSQL_PW || env.SEQ_PW || 'Password12!',
    // no port!
    // ...etc
  },
  oldest: {
    port: env.SEQ_MSSQL_PORT || env.SEQ_PORT || 22_020,
  },
  latest: {
    port: env.SEQ_MSSQL_PORT || env.SEQ_PORT || 22_019,
  },
},
```

It's just an example, the values are probably not correct. Key points:

- `native` is not desired, the docker images are the way to go as they reflect what will be used on the CI (but if really needed can be set manually by changing the `SEQ_PORT` env variable).
- Use `oldest` and `latest` instead of `2019`, `2017` to reduce how much boilerplate must be updated when we upgrade